lib/vanagon/engine/hardware.rb

```diff
@@ -0,0 +1,102 @@
+require 'vanagon/engine/base'
+require 'json'
+require 'redis'
+
+REDIS_HOST = 'redis.delivery.puppetlabs.net'
+REDIS_PORT = 6379
+LOCK_USER = "vanagon/#{ENV['USER']}"
+
+
+class Vanagon
+  class Engine
+    # Class to use when building on a hardware device (e.g. AIX, Switch, etc)
+    #
+    class Hardware < Base
+
+      # This method is used to obtain a vm to build upon
+      # For the base class we just return the target that was passed in
+      def select_target
+        @build_host
+      end
+
+      # Steps needed to tear down or clean up the system after the build is
+      # complete. In this case, we'll attempt to unlock the hardware
+      def teardown
+        release_lock
+      end
+
+      def initialize(platform)
+        # see if we can aquire a lock
+        Vanagon::Driver.logger.debug "Hardware engine invoked."
```

Comment:
I don't think this message actually belongs in the vanagon_hosts logfile. However, a message in select_target would be appropriate. Then the message could be something like:

`Vanagon::Driver.logger.info "Locked and checked out #{@build_host}"`

Then the teardown could say something about unlocking and returning `@build_host`.